flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/data/conversion/ArrayObjectArrayConverter.java

```diff
@@ -100,7 +100,6 @@ public E[] toExternal(ArrayData internal) {
             if (genericArray.isPrimitiveArray()) {
                 return genericToJavaArrayConverter.convert((GenericArrayData) internal);
             }
-            return (E[]) genericArray.toObjectArray();
```

Comment:
I'm wondering whether it's a good idea to always remove this line and then fall back  to `toJavaArray(internal)`. 
From my side, `(E[]) genericArray.toObjectArray()` seems a optimization compared to `toJavaArray(internal)`.
In the vectorized way,  it should fall to `toJavaArray(internal)`, otherwise, it will fail.
But in the non-vectorized way, everything is ok even though we don't remove this line.
For example, if we try to make array read to be non-vectorized by modifying the method `isVectorizationUnsupported`. The test will pass if we don't remove this line.